src/Core/IThreadHandling.cs

```diff
@@ -53,6 +53,16 @@ public interface IThreadHandling
         /// </summary>
         Task RunOnUIThread(Action op);
 
+        // TODO: renaming async methods to end with "Async", then rename this method to "RunOnUIThread"
+        // See https://github.com/SonarSource/sonarlint-visualstudio/issues/4180
+        /// <summary>
+        /// Executes the operation synchronously on the main thread.
+        /// If the caller is on the main thread already then the operation is executed directly.
+        /// If the caller is not on the main thread then the method will switch to the main thread,
+        /// then resume on the caller's thread when then the operation completes.
+        /// </summary>
+        void RunOnUIThreadSync(Action op);
```

Comment:
Some of the existing code uses our old non-mockable `RunOnUIThread` class directly.
I've added a new abstraction to `IThreadHandling` so I don't have to do that.
This time, however, I created a couple of separate tickets for the additional work that is required, rather than trying to fix everything at once.

I will need the new method when initialising the new path provider.